### PR TITLE
migrate to syn 2

### PR DESCRIPTION
**Pull request by B-Reif**
Updates `syn` version to `"2"` and updates usage.

Closes https://github.com/teoxoy/encase/issues/36.

### CODE REVIEW

**Comment by B-Reif** [comment]
It looks like the failing tests are being compared to older `.stderr` files which contain errors related to deriving `ShaderType`:

Currently in `invalid_align_attr.stderr`:
```
error: expected attribute arguments in parentheses: #[align(...)]
 --> tests/compile_fail/invalid_align_attr.rs:5:10
  |
5 | #[derive(ShaderType)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `ShaderType` (in Nightly builds, run with -Z macro-backtrace for more info)
```

This error isn't getting thrown in the new test result, which seems like the correct behavior? Let me know if anything needs to change here.

File: derive/impl/src/lib.rs [hunk 1]
```diff
@@ -197,15 +197,15 @@ pub fn derive_shader_type(input: DeriveInput, root: &Path) -> TokenStream {
                 align: None,
             };
             for attr in &field.attrs {
-                let span = attr.tokens.span();
-                if attr.path.is_ident("align") {
+                let span = attr.to_token_stream().span();
+                if attr.path().is_ident("align") {
```
Comment (teoxoy) [inline derive/impl/src/lib.rs:201]: I don't think this is correct and is why we seem to lose the span info.

We should use the new `attr.meta.require_list()` and work with `MetaList`s `.path` and `.tokens.span()`.

**Review by teoxoy — CHANGES_REQUESTED** [review]
Thank you for the PR!

We shouldn't have to update the expected error files; we seem to lose the span info now.

**Comment by B-Reif** [comment]
Having some trouble compiling on my machine now, will resume fixing this later

**Review by teoxoy — APPROVED** [review]

**Comment by teoxoy** [comment]
@B-Reif I ended up fixing the remaining issues since I wanted to release a new major version with the upgrade to syn2 included.

**Pull request merged**